test/integration/parachain/staging/sequential/replace.test.ts

```diff
@@ -102,7 +119,38 @@ describe("replace", () => {
             assert.equal(requestsList[0].btcAddress, firstMapEntry.value.btcAddress);
             assert.equal(requestsList[0].amount.toString(), firstMapEntry.value.amount.toString());
             assert.equal(requestsList[0].btcHeight.toString(), firstMapEntry.value.btcHeight.toString());
-        }).timeout(400000);
+        }).timeout(1000000);
+
+        it("should fail vault replace request if not having enough tokens", async () => {
+            const replaceAmount = dustValue;
+
+            interBtcAPI.setAccount(vault_2);
+
+            // check precondition: vault does not hold enough issued tokens to request a replace
+            const tokensInVault = await interBtcAPI.vaults.getIssuedAmount(
+                newAccountId(api, vault_2.address),
+                tickerToCurrencyIdLiteral(collateralCurrency.ticker)
+            );
+
+            // vault 2 should have 0 issued tokens, but tests added later may interfere...
+            // just double check here that we don't have enough to match the replace request.
+            assert.isAbove(
+                replaceAmount.add(feesEstimate).toBig(wrappedCurrency.base).toNumber(), 
+                tokensInVault.toBig(wrappedCurrency.base).toNumber(), 
+                "Pre-condition failed: vault needs fewer tokens than replace request"
+            );
+
+            try {
+                await interBtcAPI.replace.request(
+                    replaceAmount, 
+                    currencyIdToMonetaryCurrency(vault_3_id.currencies.collateral) as CollateralCurrency
+                );
+                assert.fail("Expected error to be thrown due to lack of issued tokens for vault, but call completed.");
+            } catch (e) {
+                assert.isTrue(e instanceof Error, "Expected replace request to fail with Error");
```

Comment:
Once you refactor to the builder pattern, maybe a better way to do it is to `sendLogged` with the last parameter set to the `ExtrinsicFailed` event.